src/am_map.c

```diff
@@ -955,9 +953,9 @@ boolean AM_Responder
 
       switch (automapoverlay)
       {
-        case 2:  plr->message = "Dark Overlay On";  break;
-        case 1:  plr->message = s_AMSTR_OVERLAYON;  break;
-        default: plr->message = s_AMSTR_OVERLAYOFF; break;
+        case 2:  doomprintf(MESSAGES_NONE, "Dark Overlay On");        break;
```

Comment:
Strcitly speaking, the second argument isn't a format string here. Better use the `"%s", string` syntac consistently.

Reply:
Do we have to write like this? Is `printf("string");` illegal these days?

Reply:
We explicitly declare it with `PRINTF_ATTR(2, 3)` and I think this "raises compiler awareness". I know for sure that its illegal to pass a char pointer in that case, maybe it's less strict with string literals.

Reply:
Alright. To make sure we catch all potential format string errors, I was going to add `-Wformat=2` to our compiler flags - only to find out that it is already there. So, in this case, everything that doesn't emit a warning is valid. I would appreciate, though, if you could decide for one style or the other - and probably it even makes more sense to decide for the simpler solution.

Reply:
By "one style or the other", do you mean `doomprintf(MESSAGES_NONE, "string")` vs. `doomprintf(MESSAGES_NONE, "%s", string)`?

Reply:
Yes. And apparently I was mistaken and the former is safe for string literals, so that should be the preferred format.

Reply:
Ya, the only concern is if you do something like `printf(variable)`, where you can't guarantee there aren't arguments in the string.

Reply:
`AMSTR_OVERLAYON` is a macro (i.e. a string literal), `s_AMSTR_OVERLAYON` is a `char *` pointing to that macro.

Reply:
I think you are already doing everything right. `printf("string literal");` and `printf("%s", string_var);`

Reply:
Then it was most probably me who mixed things up this morning, sorry for the noise!